evol/logger.py

```diff
@@ -3,27 +3,32 @@
 import json
 import logging
 import sys
+import uuid
 
 class BaseLogger():
     """
-    The `evol.BaseLogger` is the most basic logger in evol. You can add it to a population
-    so that the population knows how to handle the `.log()` verb. 
+    The `evol.BaseLogger` is the most basic logger in evol. 
+    You can supply it to a population so that the population 
+    knows how to handle the `.log()` verb. 
     """
-    def __init__(self, file='/tmp/evol-logs.log', stdout=False, format='%(asctime)s,%(message)s'):
+    def __init__(self, file=None, stdout=False, format='%(asctime)s,%(message)s'):
         self.file = file
         if file is not None:
             if not os.path.exists(os.path.split(file)[0]):
                 raise RuntimeError(f"path to file {os.path.split(file)[0]} does not exist!")
-        self.logger = logging.getLogger(name=self.__class__.__name__)
         formatter = logging.Formatter(fmt=format, datefmt='%Y-%m-%d %H:%M:%S')
-        if file:
-            file_handler = logging.FileHandler(filename=file)
-            file_handler.setFormatter(fmt=formatter)
-            self.logger.addHandler(file_handler)
-        if stdout:
-            stream_handler = logging.StreamHandler(stream=sys.stdout)
-            stream_handler.setFormatter(fmt=formatter)
-            self.logger.addHandler(stream_handler)
+        self.logger = logging.getLogger(name=f"{uuid.uuid4()}")
```

Comment:
i added a `uuid` because the alternative __class__.__name__ would cause errors if two logs were created in the same file. am open to better implementations.

Reply:
This works for now. We may include an optional `name` parameter for a `Population` and use that to distinguish if present.